src/utils/types.ts

```diff
@@ -1 +1,19 @@
+import { z } from "zod";
+import { Types } from "mongoose";
+
+export const commentSchema = z.object({
+    author: z.instanceof(Types.ObjectId),
+    post: z.instanceof(Types.ObjectId),
+    date: z.date().default(new Date()),
+    content: z.string(),
+    likes: z.number().default(0),
+    replyTo: z.instanceof(Types.ObjectId).nullable().default(null),
+});
+
+export const commentLikeSchema = z.object({
+    comment: z.instanceof(Types.ObjectId),
+    user: z.instanceof(Types.ObjectId),
+    date: z.date().default(new Date()),
```

Comment:
Same change with comment like schema
```suggestion
    comment: z.string().transform(id => new Types.ObjectId(id)),
    user: z.string().transform(id => new Types.ObjectId(id)),
    date: z.date().default(new Date()),
```